pkg/pq/priority_queue.go

```diff
@@ -51,16 +51,38 @@ func (pq *PriorityQueue) Len() int {
 }
 
 func (pq *PriorityQueue) Release(value Value) {
-	queue := &internalQueue{}
-	heap.Init(queue)
+	idx := pq.findIdx(func(item *Item) bool {
+		return item.value == value
+	})
 
-	for _, item := range *pq.queue {
-		if item.value != value {
-			heap.Push(queue, item)
-		}
+	if idx == -1 {
+		return
 	}
 
-	pq.queue = queue
+	// Item to replace
+	lastIdx := (*pq.queue).Len() - 1
+
+	lastItem := (*pq.queue)[lastIdx]
+	lastItem.index = idx
+
+	(*pq.queue)[idx] = lastItem
+	heap.Fix(pq.queue, idx)
+
+	(*pq.queue)[lastIdx] = nil
+	(*pq.queue) = (*pq.queue)[:lastIdx]
+
+	if (*pq.queue).Len() == 0 {
+		pq.queue = &internalQueue{}
+	}
```

Comment:
thank you for your review!
Now that I think about it..! It wouldn't need to use the following cases.
>  it creates a new memory queue when the queue size reaches zero


But isn't it necessary in this case to prevent memory leaks??
`(*pq.queue)[lastIdx] = nil`

https://github.com/golang/go/wiki/SliceTricks#delete-without-preserving-order

Reply:
`(*pq.queue)[lastIdx] = nil`
Oh! I didn't know about this. It seems correct to assign `nil` to the value of the last index. 
Thanks for correcting my mistakes! :)

Reply:
After removing unnecessary parts, I will request a review again :)

Reply:
@ppeeou 
I think @mojosoeun gave us a better suggestion, so this suggestion can be closed! :)